### ISSUE TITLE
NEWINSTALL file unlink issue and welcome.w2p creation

### ISSUE THREAD

**Issue by BuhtigithuB** [web2py/web2py#1911]
The initial report start here in this thread : 
https://groups.google.com/d/msg/web2py/tZHHKyRz0Cg/u-U9aL35BgAJ

It seems that if someone clone web2py from the Github there is not NEWINSTALL file in web2py/ root folder which log an error as web2py can unlink('NEWINSTALL') here : 
https://github.com/web2py/web2py/blob/fb4c114d85494ad6d79d69539adaae103d76e5ce/gluon/fileutils.py#L270

**Comment by BuhtigithuB** [comment]
I would solve this by tracking an "NEWINSTALL" file in the version control at the root of web2py/ folder.

It make sens as we want that web2py appadmin create a new app base on the last up to date version of the welcome app template. So when ever you clone or fetch the git repo the NEWINSTALL file will get recreated and web2py refresh the welcome.w2p app package to make sure that we create new app base on the most up to date welcome app template...

What do you think?

**Comment by cccaballero** [comment]
It could work, but users must be warned about not write code in welcome app because they can lost it. The other thing is that the file is removed after the first run, so there will be always uncommitted changes.

I need to take a second look, but the code is inside:

    if not os.path.exists('welcome.w2p') or os.path.exists('NEWINSTALL'):

It will only run if there is no a 'welcome.w2p' package or if there is a 'NEWINSTALL' file, so, maybe is not a big deal.

**Comment by BuhtigithuB** [comment]
I don't understand... As far as I understand, the only purpose of create_welcome_w2p() is to create a welcome.w2p package that will later serve as a template for creating new app from appadmin...

It not create welcome app in applications folder, this app come preinstall with web2py...

**Comment by cccaballero** [comment]
You are right, I was misunderstanding the thing, but there is the issue with the uncommitted changes after removing the file

**Comment by BuhtigithuB** [comment]
Yes this will occur... I am not sure how it would behave, but I guess it
could became pretty annoying from the version control perspective...

We could maybe make it smater... Assuming that the only purpose of this
function is what I explained... We could an unversionned file that content
an hash and a timestamp (maybe) and the version of web2py (copy/paste from
web2py version file). So we could check for version and hashes of
welcome.w2p and welcome/ as welcome.w2p and compare all of this, so if the
hash has change it may because you update locally the welcome app and you
want a new welcome.w2p package or if the version change you have to update
the welcome.w2p because welcome app surely have changed.

??

**Comment by cccaballero** [comment]
We really need to work in something so elaborated? 
I mean, maybe we are overcomplicating the issue.
The create_welcome_w2p() function is called when the framework console() starts, and it will generate a new "welcome.w2p" file if there is none or if there is a "NEWINSTALL" file.
So, the "welcome.w2p" will not be regenerating in every start because it already exists after the first run (that's why the error message is only triggered in the first run).
Right now, in my opinion, the issue is that there is a call to os.unlink('NEWINSTALL') without checking if the "NEWINSTALL" file exists.

I think that looking for a way of regenerating the "welcome.w2p" file when there are changes in the welcome app is a whole new feature.

**Comment by BuhtigithuB** [comment]
You probably right about the feature thing at the same time to me this
web2py package should be in synch no matter what way you pick to deploy you
upgraded web2py version... But if we add a try like you say it gonna work
and it will work with the web2py build (when you download from the
web2py.com site)...

Another option is to just change the level of logging and change the
message... We set the message as Warning and set a message like : "Creating
the welcome.w2p package failed you migth be unable to create from admin app
or the app you create may not be based on the last version of the welcome
app"

I leave you decide and will see what's Massimo will do in the end...

:)

**Comment by BuhtigithuB** [comment]
Wouldn't make it more sens if we get rid of NEWINSTALL file while making a
simple check when we create a new app from admin?? We check if welcome.w2p
is there if not it a new installation we create it before creating the
app... That way someone who install form web2py.com build or update from
github would work as both would remove the welcome.w2p file if it there...
The file not present in the build and when you fetch/merge from master it
would remove the welcome.w2p or tell you have a change pending (I didn't
test to see what would happen)

??

**Comment by cccaballero** [comment]
Right now the NEWINSTALL file seems to be used only for this, and as you say it can be removed, but maybe in the future can be used for trigger other behaviours. Maybe @mdipierro can argue on that.

**Comment by mdipierro** [comment]
i agree. i would keep it
